apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/SyncOnboardingCompanion.tsx

```diff
@@ -479,12 +481,25 @@ const SyncOnboardingCompanion: React.FC<SyncOnboardingCompanionProps> = ({
     };
   }, [device, allowedError, handleDesyncTimerRunsOut, desyncTimeout]);
 
+  const recoverServices = useFeature("protectServicesDesktop");
+  const recoverStoreId = recoverServices?.params?.protectId ?? "";
+
+  const [recoverOnboardingStatus, setRecoverOnboardingStatus] = useState<boolean>();
+
+  const getRecoverOnboardingStatus = useCallback(() => {
+    const userId = getUserId();
+    const status = getStoreValue(`USER_RESTORE_ONBOARDING_STATUS-${userId}`, recoverStoreId);
```

Comment:
Hi @cksanders, just to be sure I understand correctly what `USER_RESTORE_ONBOARDING_STATUS-${userId}` is used for, it's true when the recover onboarding has been successfully completed and it's false otherwise ?

Reply:
We need to share state between Live Apps, namely LL & recover. 

This wallet API storage variable is related to a user wallet id.

It initially has a value of `undefined`, then `false`, then only `true` once the user has successfully restored via recover restore. 

This value ultimately ensures a user is redirected back to onboarding given they exit the Recover Restore flow prematurely. 

At the moment this isn't happening and rather the user is being redirected to the My Ledger (the last navigation item in the menu list), as onboarding is mistakenly marked as complete, due to an optimist update given the user came from recover restore flow.

Reply:
Is it different than when `SUBSCRIPTION_STATE` is set to `BACKUP_DONE` ? (it is used in the recover banner to determine the current step of the user's onboarding)

Reply:
Yes, this is different to SUBSCRIPTION_STATE, the scope of this PR is only related to the PRs in the ticket description.